test/passes/memory-packing_all-features.wast

```diff
@@ -1,38 +1,472 @@
 (module
   (import "env" "memory" (memory $0 2048 2048))
   (import "env" "memoryBase" (global $memoryBase i32))
+  ;; nothing
+)
+
+(module
+  (import "env" "memory" (memory $0 2048 2048))
+  (import "env" "memoryBase" (global $memoryBase i32))
+  (data (i32.const 4066) "") ;; empty
+)
+
+(module
+  (import "env" "memory" (memory $0 2048 2048))
+  (import "env" "memoryBase" (global $memoryBase i32))
+
   (data (global.get $memoryBase) "waka this cannot be optimized\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00we don't know where it will go")
+
   (data (i32.const 1024) "waka this CAN be optimized\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00we DO know where it will go")
+
   (data (i32.const 2000) "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00zeros before")
+
   (data (i32.const 3000) "zeros after\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00")
+
   (data (i32.const 4000) "zeros\00in\00the\00middle\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00nice skip here\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00another\00but no")
+
+  (func $nonzero-size-init-of-active-will-trap
+    (memory.init 0
+      (i32.const 42)
+      (i32.const 0)
+      (i32.const 13)
+    )
+    (data.drop 0)
+  )
+
+  (func $nonzero-offset-init-of-active-will-trap
+    (memory.init 0
+      (i32.const 42)
+      (i32.const 13)
+      (i32.const 0)
+    )
+    (data.drop 0)
+  )
+
+  (func $zero-offset-size-init-of-active-may-trap
+    (memory.init 0
+      (i32.const 42)
+      (i32.const 0)
+      (i32.const 0)
+    )
+    (data.drop 0)
+  )
 )
+
 (module
   (import "env" "memory" (memory $0 2048 2048))
-  (import "env" "memoryBase" (global $memoryBase i32))
-  ;; nothing
+
+  (data passive "not referenced, delete me") ;; 0
+
+  (data passive "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00zeroes at start") ;; 1
+
+  (func $zeroes-at-start
+    (memory.init 1
+      (i32.const 0)
+      (i32.const 0)
+      (i32.const 45)
+    )
+    (data.drop 1)
+  )
+
+  (data passive "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00zeroes at start") ;; 2
```

Comment:
Why can't we remove zeroes at start on this segment? Is that because the threshold is higher because of more `memory.init`s? Then maybe better to add one-line comment or something to explain that.

Reply:
Exactly. I'll add a comment.